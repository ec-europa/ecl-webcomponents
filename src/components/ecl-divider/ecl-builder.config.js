const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-divider-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-divider-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-divider-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-divider-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
