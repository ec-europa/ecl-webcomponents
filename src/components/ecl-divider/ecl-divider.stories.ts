const getArgTypes = () => ({
  color_mode: { table: { disable: true } },
});

export default {
  title: 'Components/divider',
};

const Template = args =>
  `<ecl-divider theme="${args.theme}"></ecl-divider>`;

export const divider = Template.bind({});
divider.storyName = 'default';
divider.argTypes = getArgTypes();
