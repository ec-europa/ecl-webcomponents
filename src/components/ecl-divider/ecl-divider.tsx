import { Component, h, Prop, Element } from '@stencil/core';

@Component({
  tag: 'ecl-divider',
  styleUrls: {
    ec: './build/styles/ecl-divider-ec.css',
    eu: './build/styles/ecl-divider-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclDivider {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;

  getClass(): string {
    const styleClasses = [
      `ecl-separator`,
      this.styleClass
    ];

    return styleClasses.join(' ');
  }

  render() {
    return (
      <hr class={this.getClass()} />
    );
  }
}