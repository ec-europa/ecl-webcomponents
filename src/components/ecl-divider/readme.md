# ecl-divider



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default     |
| ------------ | ------------- | ----------- | -------- | ----------- |
| `styleClass` | `style-class` |             | `string` | `undefined` |
| `theme`      | `theme`       |             | `string` | `'ec'`      |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
