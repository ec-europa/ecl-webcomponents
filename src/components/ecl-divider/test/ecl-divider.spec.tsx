import { newSpecPage } from '@stencil/core/testing';
import { EclDivider } from '../ecl-divider';

describe('ecl-divider', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclDivider],
      html: `<ecl-divider theme="ec"></ecl-divider>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});

