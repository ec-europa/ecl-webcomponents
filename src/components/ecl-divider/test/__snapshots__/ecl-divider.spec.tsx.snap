// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-divider renders 1`] = `
<ecl-divider theme="ec">
  <hr class="ecl-separator">
</ecl-divider>
`;
